README.rst

```diff
@@ -939,6 +1009,194 @@ configuration status of a device changes, this ensures the check reacts
+
+By default iperf check runs without any kind of **authentication**, But we can configure it to use **RSA authentication** 
+between the **client** and the **server** to restrict the connections to the server & only allow legitimate clients.
+
+At Iperf server
+###############
+
+1. Generate RSA keypair
+^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: shell
+
+   openssl genrsa -des3 -out private.pem 2048
+   openssl rsa -in private.pem -outform PEM -pubout -out public.pem
+   openssl rsa -in private.pem -out private_not_protected.pem -outform PEM
+
+After running above mention commands, the public key will be contained in the 
+file ``public.pem`` which will be used in **rsa_public_key** parameter 
+in `OPENWISP_MONITORING_IPERF_CHECK_CONFIG <#OPENWISP_MONITORING_IPERF_CHECK_CONFIG>`_ 
+and the private key will be contained in the file ``private_not_protected.pem`` 
+which will be used with **--rsa-private-key-path** command option at iperf server.
+
+2. Create user credentials
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: shell
+
+   USER=iperfuser PASSWD=iperfpass
+   echo -n "{$USER}$PASSWD" | sha256sum | awk '{ print $1 }'
+   ----
+   ee17a7f98cc87a6424fb52682396b2b6c058e9ab70e946188faa0714905771d7 #This is the hash of "iperfuser"
+
+Add the above hash with username in ``credentials.csv``
+
+.. code-block:: shell
+
+   # file format: username,sha256
+   iperfuser,ee17a7f98cc87a6424fb52682396b2b6c058e9ab70e946188faa0714905771d7
+
+3. Now start the iperf server with auth options
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: shell
+
+   iperf3 -s --rsa-private-key-path ./private_not_protected.pem --authorized-users-path ./credentials.csv
+
+At client (openwrt device)
+##########################
+
+1. Install iperf3-ssl
+^^^^^^^^^^^^^^^^^^^^^
+
+Install `iperf3-ssl openwrt package <https://openwrt.org/packages/pkgdata/iperf3-ssl>`_ instead of normal
+`iperf3 openwrt package <https://openwrt.org/packages/pkgdata/iperf3>`_ which comes without any authentication.
+
+You may also check your installed **iperf3 openwrt package** features:
+
+.. code-block:: shell
+
+   root@vm-openwrt:~ iperf3 -v
+   iperf 3.7 (cJSON 1.5.2)
+   Linux vm-openwrt 4.14.171 #0 SMP Thu Feb 27 21:05:12 2020 x86_64
+   Optional features available: CPU affinity setting, IPv6 flow label, TCP congestion algorithm setting, 
+   sendfile / zerocopy, socket pacing, authentication # contains 'authentication'
+
+2. Configure iperf check auth parameters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now, add the following iperf authentication parameters 
+to `OPENWISP_MONITORING_IPERF_CHECK_CONFIG <#OPENWISP_MONITORING_IPERF_CHECK_CONFIG>`_ 
+in `openwisp settings <https://github.com/openwisp/openwisp-monitoring/blob/master/tests/openwisp2/settings.py>`_
+
+.. code-block:: python
+
+    OPENWISP_MONITORING_IPERF_CHECK_CONFIG = {
+    # All three parameters are required
+    'username': {'default': 'iperfuser'},
+    'password': {'default': 'iperfpass'},
```

Comment:
what is `default`?

Reply:
I made it to look consistent with [`ping config override setting`](https://github.com/openwisp/openwisp-monitoring#openwisp_monitoring_ping_check_config), Now I have changed `iperf config setting` according to https://github.com/openwisp/openwisp-monitoring/issues/418